server/mqtt.go

```diff
@@ -2833,6 +2959,30 @@ func (as *mqttAccountSessionManager) transferRetainedToPerKeySubjectStream(log *
 	return nil
 }
 
+func (as *mqttAccountSessionManager) getCachedRetainedMsg(subject string) *mqttRetainedMsg {
+	v, ok := as.rmsCache.Load(subject)
+	if !ok {
+		return nil
+	}
+	rm := v.(*mqttRetainedMsg)
```

Comment:
I think that there is still a place where you cast to `(mqttRetainedMsg)` without the pointer. That would be in `cleaupRetainedMessageCache()`. Not sure if Go does the right thing, but for consistency, I would change the casting there.

Reply:
fixed, thx! not sure how to add a test for the cleanup, would've been nice.